### ISSUE TITLE
Open tasks feature: screenshots and discussion [code almost complete]

### ISSUE THREAD

**Issue by jonkerz** [calacademy-research/antcat#130]
**TL;DR**: A task list where editors can add, comment on, and tick off completed tasks.

**Problem**: Besides email, editors have no way of coordinating changes and making other editors aware of what needs to be fixed. Email has always worked, but it has never been great. It's hard to follow long email conversations, and hard to know if a particular change was made or not. And if a suggested change cannot be made quickly, it's likely to be forgotten. Also, if we fail to reply to user feedback, no-one will bother submitting corrections in the future.

**Solution**: A feature where editors can add "open tasks" that are then shown in a list in the GUI. Items can be commented on, and ticked off as completed or closed.* Hopefully to be expanded so that any user, and not just editors, can create new tasks and add comments.

**Counter argument**: Writing custom code for this, as opposed to using existing third-party issue trackers, would be re-inventing the wheel (read: stupid) and it would never be as good as already existing software.

**Counter counter arguments**:
* We have a lot to gain by implementing this ourselves, examples:
  * In the task descriptions, we can use a simple format to link taxa and references (which an external tool wouldn't have knowledge of).
  * Show "This taxon is linked from an open tasks" on catalog pages.
  * Displaying the number of open tasks in the top header.
  * In the future: send email notifications to a set of editors (picked from a list) after creating the new task, saying that there's an open task that they may be able to help solve.
  * In the super future: create cool hooks such as automatically emailing editors who have added _eg_ _Lasius_ to their "watchlist". 
* Getting all users to sign up for an external tool will just not happen.
* It ties in well with the upcoming activity feed (which also would not be possible using external tools). Example, with the most recent events first:
  * `Brian Fisher completed the task "Update Atta per M. R Dangerous (2016)" 4 hours ago`
  * `Brian Fisher edited the genus Atta 4 hours ago`
  * `Brian Fisher added the new species Atta peligrosa 5 hours ago`
  * `Flavia Esteves added the new open task "Update Atta per M. R Dangerous (2016)" 2 days ago`

**More**:
* I'm working on some scripts that scan the database for errors such as synonyms listed as valid, etc.; adding these to the task list for editors to review is another good use-case.

*on the possibly confusing distinction between completed/closed:
* "Completed" = the change(s) mentioned in the task has been made, case closed.
* "Closed" = the change(s) were not made, case closed.
* We want to make this distinction because say someone adds the task `"Change the spelling of Formicidae to Formichidee"` -- when this task is closed, we want the activity feed to say `Quintus Batiatus closed the task "Change the spelling of Formicidae to Formichidee"` not that is was "completed", which would confuse editors.

**Comment by jonkerz** [comment]
Screenshot from dev:
![01_feed](https://cloud.githubusercontent.com/assets/1513381/14361409/012e3cd2-fcfb-11e5-9fe4-953e3c63608a.png)

TODO list implemented events/actions.

**Comment by bpescador** [comment]
@jonkerz  I am in favor of your suggestion of having a TL.  We have had a number of scripts pull up errors in AntCat but we have lost track of the status of fixing those errors.  Having a clear summary of what needs to be done would be very helpful.

**Comment by jonkerz** [comment]
@bpescador
> We have had a number of scripts pull up errors

More than the ones listed in #6? I've added a bunch of new snippets there, and collected others found in the repo.

I could add a field for categorizing this type of data tasks, to make it easier to find closed tasks that may have to be re-run once in a while.
